lib/cosmos/packets/packet.rb

```diff
@@ -209,6 +210,20 @@ def identify?(buffer)
 
     end
 
+    # Calculates a unique MD5Sum that changes if the parts of the packet configuration change that could affect
+    # the "shape" of the packet.  This value is cached and that packet should not be changed if this method is being used
```

Comment:
Does the ALLOW_SHORT keyword change the "shape" of the packet?

Reply:
No, ALLOW_SHORT only affects if an undersized buffer is allowed to be assigned to a packet without raising an exception (the rest get filled with zeros).   The shape of the packet is the same either way.